Assets/Scripts/Algorithms/Patrolling/CognitiveCoordinated/CognitiveCoordinatedVirtualStigmergy.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections.Generic;
+using System.Linq;
+
+using Maes.Algorithms.Patrolling.Components;
+using Maes.Map;
+using Maes.Robot;
+
+namespace Maes.Algorithms.Patrolling
+{
+    /// <summary>
+    /// This is a distributed version of the Cognitive Coordinated Algorithm.
+    /// It uses virtual stigmergy to coordinate the robots.
+    /// Each robot has its own local map and uses the occupied vertices of other robots to coordinate.
+    /// </summary>
+    public sealed class CognitiveCoordinatedVirtualStigmergy : CognitiveCoordinatedBase
+    {
+        public CognitiveCoordinatedVirtualStigmergy(int amountOfRobots)
+        {
+            _amountOfRobots = amountOfRobots;
+        }
+
+        private readonly int _amountOfRobots;
+        public override string AlgorithmName => "Cognitive Coordinated (virtual-stigmergy knowledge) Algorithm";
+        private VirtualStigmergyComponent<int, Vertex> _virtualStigmergyComponent = null!;
```

Comment:
Don't store Vertex directly. Each robot has its own copy of PatrollingMap with its own vertices. You are technically smuggling information here. It should be an int with the vertex id.

Reply:
Or at least the Except linq method would not work.

Reply:
Hmm.. virtual stigmergy needs the value to derive class, why is that, I cannot store/share vertex ids?

Reply:
Alright, let me fix that.